### PR TITLE
Added support for planets in modules other than core

### PR DESCRIPTION
**Pull request by BenjaminAmos**
# Description
This pull request adds support for configuring new planets in modules other than the core module.

# Testing
If the pull request works, then the default built-in planets should still function properly. It should also load a planetsConfig.json file from any other modules where it exists as well.

# Notes
This pull request also includes support for modules adding names to the planet and solar system name lists, although for planet names this is essential, as otherwise the new planets will not have any names to choose from. This can be done through the planetNamesConfig.json and systemNamesConfig.json files.

I created the [blackHoles](https://github.com/BenjaminAmos/blackHoles) module for testing this. It can be used to verify that the changes work correctly.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

File: engine/src/main/java/org/destinationsol/game/planet/PlanetConfigs.java [hunk 1]
```diff
@@ -22,11 +22,9 @@
 import org.destinationsol.files.HullConfigManager;
 import org.destinationsol.game.GameColors;
 import org.destinationsol.game.item.ItemManager;
+import org.terasology.assets.ResourceUrn;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (vampcat) [inline engine/src/main/java/org/destinationsol/game/planet/PlanetConfigs.java]: Please, no star imports.

File: engine/src/main/java/org/destinationsol/game/planet/PlanetConfigs.java [hunk 2]
```diff
@@ -40,26 +38,29 @@ public PlanetConfigs(HullConfigManager hullConfigs, GameColors cols, ItemManager
         medium = new ArrayList<>();
         hard = new ArrayList<>();
 
-        Assets.cacheLists();
+        Set<ResourceUrn> planetJsonConfigs = Assets.getAssetHelper().list(Json.class, "[a-zA-Z0-9]*:planetsConfig");
 
-        Json json = Assets.getJson("core:planetsConfig");
-        JsonValue rootNode = json.getJsonValue();
+        for (ResourceUrn planetConfigJson : planetJsonConfigs) {
+            Assets.cacheLists();
```
Comment (vampcat) [inline engine/src/main/java/org/destinationsol/game/planet/PlanetConfigs.java]: Actually, we can remove this line.

Yes, I know you did not add this, you just copied it from the existing source - but I looked into the details, and it only caches textures, which are not being queried here.

File: engine/src/main/java/org/destinationsol/game/planet/PlanetConfigs.java [hunk 3]
```diff
@@ -40,26 +38,29 @@ public PlanetConfigs(HullConfigManager hullConfigs, GameColors cols, ItemManager
...
+                } else if (planetConfig.easyOnly) {
+                    easy.add(planetConfig);
+                } else {
+                    medium.add(planetConfig);
+                }
+            }
 
-        Assets.uncacheLists();
+            json.dispose();
+            Assets.uncacheLists();
```
Comment (vampcat) [inline engine/src/main/java/org/destinationsol/game/planet/PlanetConfigs.java]: Correspondingly, remove this line as well

File: engine/src/main/java/org/destinationsol/game/SolNames.java [hunk 4]
```diff
@@ -18,16 +18,29 @@
...
+import java.util.Set;
 
 public class SolNames {
     public final ArrayList<String> planets;
     public final ArrayList<String> systems;
 
     public SolNames() {
-        planets = readList("core:planetNamesConfig");
-        systems = readList("core:systemNamesConfig");
+
```
Comment (vampcat) [inline engine/src/main/java/org/destinationsol/game/SolNames.java]: Remove this newline

**Comment by Steampunkery** [comment]
@GooeyHub ok to test

**Comment by vampcat** [comment]
@GooeyHub : add to whitelist

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by BenjaminAmos** [comment]
The changes should have been made. I also removed the "imagePath" values from the built-in ground files, as per your suggestion, although this can be reverted if they are stilll needed.

**Comment by Steampunkery** [comment]
Can you please add a planetNamesConfig.json to your blackHoles module? The game will barf if there are no planets in core and a planets module has no names config.

**Comment by Steampunkery** [comment]
Also can you please change hardOnly = false into hardOnly = true in your planetsConfig file? Black holes are super deadly according to my testing.

Also, I added a new flag for planet config files called "allModes". It allows you to specify that you want that planet to be present in all game modes. I'll make a PR for it soon. The reason it's necessary is when you only have one type of planet to spawn in (an uncommon occurrence), the each planet can only belong to one game mode. So the game freaks out when there are no planets available for a given game mode. Hence the "allModes" flag. You shouldn't need it, black holes should only be in hard mode lol.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by BenjaminAmos** [comment]
I have implemented the requested changes.

The [blackHoles](https://github.com/BenjaminAmos/blackHoles) module, which I used for testing these changes, has also been updated to support the new changes.

**Pull request merged**